src/include/mca/mca.h

```diff
@@ -1009,5 +1085,111 @@ inline CalculationTaskNum threadCalculationTaskNum(const size_type &total) {
     if (total % calculation > 0) { taskNum++; }
     return CalculationTaskNum{calculation, taskNum};
 }
+
+template <class T>
+void pow(Matrix<T> &a, const size_type &exponent) {
+    assert(a.isSquare());
+    Matrix<T> output(a.shape());
+    pow(a, exponent, output);
+    a = std::move(output);
+}
+
+template <class T, class O, class>
+void pow(const Matrix<T> &a, const size_type &exponent, Matrix<O> &output) {
+    assert(a.isSquare());
+    assert(a.shape() == output.shape());
+    size_type b = exponent;
+    Matrix<T> temp(a);
+    // make output an identity matrix
+    output = Matrix<O>(output.shape(), IdentityMatrix());
+    while (b > 0) {
+        if (b & 1) { output *= temp; }
+        if ((b >> 1) == 0) { break; }
```

Comment:
If the `if` statement is remoeved, when `b` is 1, we will calculate `temp *= temp` once more which is not necessary. As a `lib`, this kind of calculation should be avoided, because the time complexity of matrix multiplication is $O(n^3)$, even just one time matrix multiplication may cost several seconds.